bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,761 @@
+                    command.getOutdoorTemperature());
+        }
+
+    }
+
+    @Override
+    public void receivedResponse(SetPowerstateCommand command) {
+
+        updateStateIfLinked(new ChannelUID(getThing().getUID(), DaikinMadokaBindingConstants.CHANNEL_ID_ONOFF_STATUS),
+                command.getPowerState());
+
+        madokaSettings.setOnOffState(command.getPowerState());
+
+        if (command.getPowerState() == OnOffType.ON) {
+            // Depending on the state
+            if (madokaSettings.getOperationMode() == null) {
+                return;
+            }
+
+            switch (madokaSettings.getOperationMode()) {
+                case AUTO:
+                    updateStateIfLinked(
+                            new ChannelUID(getThing().getUID(),
+                                    DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                            new StringType("Auto"));
+                    break;
+                case HEAT:
+                    updateStateIfLinked(
+                            new ChannelUID(getThing().getUID(),
+                                    DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                            new StringType("Heating"));
+                    break;
+                case COOL:
+                    updateStateIfLinked(
+                            new ChannelUID(getThing().getUID(),
+                                    DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                            new StringType("Cooling"));
+                    break;
+                default: // Other Modes are not [yet] supported
+                    break;
+            }
+        } else {
+            updateStateIfLinked(
+                    new ChannelUID(getThing().getUID(),
+                            DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                    new StringType("Off"));
+
+        }
+    }
+
+    @Override
+    public void receivedResponse(SetSetpointCommand command) {
+
+        // The update depends on the mode - so if not set - skip
+        if (this.madokaSettings.getOperationMode() == null) {
+            return;
+        }
+
+        switch (this.madokaSettings.getOperationMode()) {
+            case HEAT:
+                this.madokaSettings.setSetpoint(command.getHeatingSetpoint());
+                break;
+            case COOL:
+                this.madokaSettings.setSetpoint(command.getCoolingSetpoint());
+                break;
+            case AUTO:
+                // TODO
+                break;
+            default:
+                return;
+        }
+
+        if (madokaSettings.getSetpoint() != null) {
+            @NonNull
+            DecimalType dt = (@NonNull DecimalType) madokaSettings.getSetpoint();
```

Comment:
Do not use casting to get around the null checker. It should be sufficient to store it in a local variable then null check that.